src/crypto.ts

```diff
@@ -1,18 +1,30 @@
 export function isCryptoSupported(): boolean {
-  return (
-    TextEncoder !== undefined &&
-    window.crypto !== undefined &&
-    window.crypto.subtle !== undefined
-  );
+  if (typeof window === 'undefined') {
+    return true;
+  } else {
+    return (
+      TextEncoder !== undefined &&
+      window.crypto !== undefined &&
+      window.crypto.subtle !== undefined
+    );
+  }
 }
 
 export async function sha256(input: string): Promise<string> {
-  const data = new TextEncoder().encode(input);
-  const digest = await window.crypto.subtle.digest('SHA-256', data);
-  const byteArray = new Uint8Array(digest);
-  const chars = new Array(byteArray.length);
-  byteArray.forEach((byte, index) => {
-    chars[index] = byte.toString(16).padStart(2, '0');
-  });
-  return chars.join('');
+  if (typeof window === 'undefined') {
```

Comment:
This condition is constant throughout the entire lifetime of the app, so you can pull it out like this:
```js
let sha256;
if (typeof window === 'undefined') {
  sha256 = async (input: string): Promise<string> => {
    const crypto = require('crypto');
    // ...
  };
} else {
  sha256 = async (input: string): Promise<string> => {
    const data = new TextEncoder().encode(input);
    // ...
  };
}

export sha256;
```